### DISCUSSION TITLE
对项目二攻击范围的疑问

### DISCUSSION THREAD

**Discussion by ghost** [liusy58/NJU_AdvancedProgramming_SP22#179] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/100082486/165117692-36a32119-6197-4972-beb7-5dec7b3e3b9e.png)
远程敌人的攻击范围严格大于1，指的是保证`AttackRange>1`，当`distance<=AttackRange`时开始攻击。还是`1<distance<AttackRange`时开始攻击？

![image](https://user-images.githubusercontent.com/100082486/165118212-40bddeb9-82a2-4e0b-bb0b-eaca4f3dc09e.png)
近战敌人的攻击范围是自己的所在格子，而近战单位又有阻隔作用，所以在近战敌人进入近战单位所在的格子前，都无法攻击近战单位？
换句话讲，近战敌人可以攻击某格的近战单位，当且仅当近战单位起到阻隔作用或者近战单位阻隔上限达到且敌人还未离开这一格？

另外，距离必须是欧氏距离吗？能否用切比雪夫距离？

**Comment by liusy58** [comment, marked as answer]
三个问题的回答都是都可以，其实第二个项目没有什么很硬性的规定，你认为合理就可以。
